test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/irrule/IRRule.java

```diff
@@ -44,10 +45,10 @@ public class IRRule implements Matchable {
     private final IR irAnno;
     private final MatchableMatcher matcher;
 
-    public IRRule(int ruleId, IR irAnno, Compilation compilation) {
+    public IRRule(int ruleId, IR irAnno, Compilation compilation, VMInfo vmInfo) {
         this.ruleId = ruleId;
         this.irAnno = irAnno;
-        this.matcher = new MatchableMatcher(new CompilePhaseIRRuleBuilder(irAnno, compilation).build());
+        this.matcher = new MatchableMatcher(new CompilePhaseIRRuleBuilder(irAnno, compilation).build(vmInfo));
```

Comment:
For consistency with what exactly? How would I pass it on/down from the `CompilePhaseIRRuleBuilder` constructor?

Reply:
In the IR framework, I'm usually passing the required information to the constructor of the builder class and then just call `build()`. You could also do the same here since you otherwise need to pass the `vmInfo` around inside the `CompilePhaseIRRuleBuilder` class to the different private methods.